clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -599,14 +605,12 @@ private void clear() {
             Deque<NetworkReceive> deque = this.stagedReceives.get(channel);
             boolean sendFailed = failedSends.remove(channel.id());
             if (deque == null || deque.isEmpty() || sendFailed) {
+                channel.state(ChannelState.FAILED_SEND);
                 doClose(channel, true);
                 it.remove();
             }
         }
         for (String channel : this.failedSends) {
-            KafkaChannel failedChannel = closingChannels.get(channel);
```

Comment:
It seems like this can never happen because we we remove the channel id from `failedSends` in the loop where we iterate over the closing channels.